app/uk/gov/hmrc/vatregisteredcompanies/repositories/VatRegisteredCompaniesRepository.scala

```diff
@@ -51,47 +53,80 @@ object Wrapper {
 }
 
 @Singleton
-class   VatRegisteredCompaniesRepository @Inject()(reactiveMongoComponent: ReactiveMongoComponent)(implicit val executionContext: ExecutionContext)
-  extends ReactiveRepository("vatregisteredcompanies", reactiveMongoComponent.mongoConnector.db, Wrapper.format) {
-
-  def processList(bd: List[PayloadWrapper]): Future[Unit] = {
-    for {
-      _ <- bd.map(x => process(x.payload))
-    } yield (())
-    Future.successful(())
-  }
+class   VatRegisteredCompaniesRepository @Inject()(reactiveMongoComponent: ReactiveMongoComponent)
+  (implicit val executionContext: ExecutionContext) extends
+  ReactiveRepository("vatregisteredcompanies", reactiveMongoComponent.mongoConnector.db, Wrapper.format) {
 
   implicit val format: OFormat[Wrapper] = Json.format[Wrapper]
 
-  private def update(entry: Wrapper): Future[Unit] = {
+  private def insert(entries: List[Wrapper]): Future[Unit] = {
+    bulkInsert(entries).map(_ => (()))
+  }
 
-    domainFormatImplicit.writes(entry) match {
-      case a@JsObject(_) =>
-        val selector = Json.obj("vatNumber" -> entry.vatNumber)
-        collection.update(selector, entry)
-      case _ =>
-        Future.failed[WriteResult](new Exception("failed insert or update"))
+  private def delete(deletes: List[VatNumber]): Future[Unit] = {
+    val bulkSize = ProtocolMetadata.Default.maxBulkSize - 1
+    val it = deletes.sliding(bulkSize,bulkSize)
+    while (it.hasNext) {
+      val chunk = it.next
+      val deleteBuilder = collection.delete(false)
+      val ds: Future[List[collection.DeleteCommand.DeleteElement]] =
+        Future.sequence(
+          chunk.map { vatNumber =>
+            deleteBuilder.element(
+              q = BSONDocument("vatNumber" -> vatNumber),
+              limit = None,
+              collation = None)
+          }
+        )
+      ds.flatMap { ops =>
+        deleteBuilder.many(ops)
+      }.map {
+        multiBulkWriteResult => multiBulkWriteResult.errmsg.foreach( e =>
+          Logger.error(s"$e")
+        )
+      }
     }
-  }.map(_ => ())
+    Future.successful((()))
+  }
 
-  private def upsert(entries: List[Wrapper]): Future[Unit] = {
-    bulkInsert(entries) map { x =>
-      x.writeErrors.foreach(e => if (e.code == 11000) {
-        entries.get(e.index).fold((()))(update)
-      })
-    }
+  private def deleteB(deletes: List[BSONValue]): Future[Unit] = {
```

Comment:
rename and limit to 1

Reply:
Renamed, not limiting as not sure it improves performance - the query will only ever return one row but we don't need to limit this.